### ISSUE TITLE
Drupal WxT Variant Install Problems

### ISSUE THREAD

**Issue by mrjmoskau** [wet-boew/wet-boew-drupal#638]
Here's the part that's still tripping me up, since the README.md and wiki instructions are different (truncated for space):

README.md > drush si wetkit wetkit_wetboew_selection_form.theme=wetkit_adaptivetheme
wiki > drush si wetkit wetkit_theme_form.theme=wetkit_adaptivetheme

Which option am I supposed to use, and where should I be when I use it?  In my testing, I'm in /srv/www/htdocs/wet_boew_drupal when I run the command, but is this correct?

This might be unrelated, but when I try to run the command on my Linux server at home, it gives me a message about the installation profile not being found. If I drop the "wetkit wetkit...adaptivetheme" parameters from the drush si command, the install goes ahead, but I don't get anything truly workable after that.

**Comment by ssari** [comment]
The three commands you need to install is in the page https://github.com/wet-boew/wet-boew-drupal - step 4 and 5.
I suggest you create the wetkit folder,  let say $webserver_root/wetkit, inside this folder execute step 4 using GitBash1.7+

git clone https://github.com/wet-boew/wet-boew-drupal.git profiles/wetkit;
drush make --prepare-install --no-gitinfofile --working-copy profiles/wetkit/build-wetkit.make --yes

please note that you might have Drush5.8+.

Step 5 you have to execute in Drush command 
drush si wetkit wetkit_wetboew_selection_form.theme=wetkit_adaptivetheme --sites-subdir=drupal_wet --db-url=mysql://drupalusr:drupalusr_pass@localhost:3306/wetkit_db --account-name=admin --account-pass=WetKit@2012 --account-mail=admin@example.com --site-mail=admin@example.com --site-name="Web Experience Toolkit"

If you don't add --sites-subdir, it will buil under sites/default
The command --db-url will create the database, just be sure you have the user drupalusr, and the password drupalusr_pass should be the same you have in my.ini.

Hope this will help you.

**Comment by sylus** [comment]
Thanks @ssari bang on!

I'll update the documentation this weekend to be a bit more clear.

**Comment by mrjmoskau** [comment]
Thanks for the clarification @ssari  I don't have my laptop with me, but I will be trying out those steps as soon as I get a chance.

I think what still throws me off is that you create a wetkit folder, and that has a profiles/wetkit folder inside it. It almost seems recursive. Also, I've found that most of the time, the "build-wetkit.make" file ends up in the top-level wetkit folder, not under profiles/wetkit, as the instructions would seem to indicate.

I'll report my findings once I've had a chance to play with this more...

**Comment by ssari** [comment]
Thanks @sylus.
If you have some time, would you please check the update commands. 
I'm getting these messages:

....\wetkit>drush make --no-gitinfofile --working-copy https://raw.github.com/wet-boew/wet-boew-drupal/master/build-wetkit.make .

file_get_contents(): Filename cannot be empty make.utilities.inc:461                       [warning]
unlink(): Invalid argument drush.inc:792                                                   [warning]
Invalid or empty make file:                                                                  [error]
https://raw.github.com/wet-boew/wet-boew-drupal/master/build-wetkit.make

**Comment by sylus** [comment]
This is how I do it from an already installed and configured install.

Basically if you want to update to the latest development version, point to your web root and just use git to update your profile checkout and then use drush to rebuild the make file and update the database for any changes defined in schema.

These two commands has worked for me (WEBROOT):

``` bash
cd profiles/wetkit && git pull && cd .. && cd .. && drush make --working-copy profiles/wetkit/build-wetkit.make --yes

drush updatedb
```

**Comment by ssari** [comment]
Something went wrong!!!

Command 1, using Git Bash
$ cd profiles/wetkit && git pull && cd .. && cd .. && drush make --working-copy profiles/wetkit/build-wetkit.make –yes
Updating 6458cfd..c2168ea ….
It worked fine with couple warnings that are showed in the installation as well.

Command 2 using Drush Bash

> drush updatedb
> Couple warnings, see attached image, such as
> version  (Currently using Unresolved dependency Variable realm (Version 7.x-2.x required))
> WetKit MetaTags Dublin Core (HTML5) requires this module and version. Currently using      [warning]

Result in the browser:
The main-content is not showed, with the Error message:
PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'base.path' in 'field list': SELECT revision.vid AS vid, revision.timestamp AS timestamp, revision.uid AS uid, revision.title AS title, revision.log AS log, revision.vuuid AS vuuid, base.fpid AS fpid, base.bundle AS bundle, base.link AS link, base.path AS path, base.reusable AS reusable, base.admin_title AS admin_title, base.admin_description AS admin_description, base.category AS category, base.view_access AS view_access, base.edit_access AS edit_access, base.created AS created, base.changed AS changed, base.uuid AS uuid, base.language AS language, base.vid AS current_vid FROM {fieldable_panels_panes} base INNER JOIN {fieldable_panels_panes_revision} revision ON revision.vid = base.vid WHERE (base.fpid IN (:db_condition_placeholder_0, :db_condition_placeholder_1, :db_condition_placeholder_2)) ; Array ( [:db_condition_placeholder_0] => 7 [:db_condition_placeholder_1] => 8 [:db_condition_placeholder_2] => 9 ) inDrupalDefaultEntityController->load() (line 196 of xx\wetkit\includes\entity.inc).

![Wetkit-update-output](https://f.cloud.github.com/assets/1461809/22255/83fd2592-49f3-11e2-8a72-9b497b387891.png)

**Comment by sylus** [comment]
Hmm it seems you have a significantly older copy and it doesn't like updating to the most recent from that.

How old is the copy you are migrating from?

The only solution I think is to finally start implementing strict tagging and doing official releases ensuring that updates work between those stable tags.

**Comment by ssari** [comment]
The one I tryied was installed in Dec 4th. I'll try the commands in the one I installed yesterday.

Any how,  specific information about the version would be great.

**Comment by sylus** [comment]
Yeah I have a few other minor issues to close up then will do an official release something like 7.x-1.0-beta1

This will be done during the holidays (I have 6 days off) so can devote time to this longstanding issue.

Also @ssari,

If you could do a fresh install on latest and then try the update functionality to see if everything works for you, that would greatly help.

I know nothing would change as the in place update would just replace the exact same files (since would only be updating on latest) but would at least help to determine if something is wrong with your setup or the update still isn't working in the most optimal of cases. Make sense?

So basically:

1) Do a fresh build + install on latest
2) Update using the requisite commands given in the readme (update will replace all files with the exact same ones since on latest anyways)
3) Does everything still work for you?

**Comment by ssari** [comment]
Yes, the update commands worked in a fresh install.

I got PDO exception to install, so I updated my.ini, and voil'a.

Just for curiosity I'll try to update again the other Wetkit in this server.

Thanks.

**Comment by DanMorand** [comment]
Just completing the Acquia Dev import and I hit a PHP error - Fatal error: Call to undefined function field_attach_load in /includes entity.inc on line 321.

**Comment by sylus** [comment]
Cleaning up some old issues. :) and for the last comment that should be fixed on RC
